### PR TITLE
Fix `desc.ref` parsing using an incorrect newline symbol.

### PR DESCRIPTION
**Pull request by serhii-lekariev**
This PR closes https://github.com/SpineEventEngine/base/issues/329.

When the `desc.ref` file is parsed, names of the referenced JARs are being split by a platform-dependent line separator, which causes build failures when the actual platform separator is different from the line separator of the platform that the `desc.ref` file is built on.

### CODE REVIEW

**Comment by serhii-lekariev** [comment]
@armiol, PTAL.

**Comment by serhii-lekariev** [comment]
@armiol, PTAL.

**Comment by serhii-lekariev** [comment]
@armiol, I have applied the changes discussed vocally, PTAL.

File: base/src/main/java/io/spine/code/proto/DescriptorReference.java [hunk 1]
```diff
@@ -82,14 +89,34 @@ public static DescriptorReference toOneFile(File file) {
...
+            Files.write(targetFile, ImmutableList.of(result), TRUNCATE_EXISTING);
+        } catch (IOException e) {
+            throw illegalStateWithCauseOf(e);
+        }
+    }
 
-        directory.toFile()
-                 .mkdirs();
+    /**
+     * A testing method that reproduces an illegal situation - a reference gets appended, but with
```
Comment (armiol) [inline base/src/main/java/io/spine/code/proto/DescriptorReference.java]: It's not a testing method. 

Please document it properly.

File: base/src/main/java/io/spine/code/proto/DescriptorReference.java [hunk 2]
```diff
@@ -82,14 +89,34 @@ public static DescriptorReference toOneFile(File file) {
...
+     * some newline symbol at the end. This should not affect extraction of resources.
+     */
+    @VisibleForTesting
+    void writeToWithSeparator(Path directory, String separator) {
```
Comment (armiol) [inline base/src/main/java/io/spine/code/proto/DescriptorReference.java]: Let's have it named `writeTo(.. , ..)` thus making it an overload.

File: base/src/main/java/io/spine/code/proto/DescriptorReference.java [hunk 3]
```diff
@@ -119,6 +152,10 @@ private static String readCatalog(URL resourceUrl) {
         }
     }
 
+    ResourceReference asResource() {
```
Comment (armiol) [inline base/src/main/java/io/spine/code/proto/DescriptorReference.java]: Please document.

I think, it's also `@VisibleForTesting`.

File: base/src/test/java/io/spine/code/proto/DescriptorReferenceTest.java [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+            result.createNewFile();
+            return result;
+        } catch (IOException e) {
+            throw Exceptions.newIllegalStateException(e,
+                                                      "Could not create a temporary file in %s.",
+                                                      path.toAbsolutePath());
+        }
+    }
+}
+
```
Comment (armiol) [inline base/src/test/java/io/spine/code/proto/DescriptorReferenceTest.java]: Let's kill this line.

File: base/src/test/java/io/spine/code/proto/given/DescriptorReferenceTestEnv.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+import io.spine.code.proto.DescriptorReference;
+import io.spine.code.proto.FileDescriptors;
+
+import java.io.File;
+import java.util.UUID;
+
+/**
+ * A utility class that provides references to descriptors.
+ */
+@SuppressWarnings("HardcodedLineSeparator")
```
Comment (armiol) [inline base/src/test/java/io/spine/code/proto/given/DescriptorReferenceTestEnv.java]: I don't see any usages of hard-coded line separators. Please re-check.

File: base/src/test/java/io/spine/io/ResourceFilesTest.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+@DisplayName("Resource files should")
+@ExtendWith(TempDirectory.class)
+class ResourceFilesTest {
+
+    @Test
+    @DisplayName("return an empty iterator if queried for a non-existing file ")
+    void throwOnInvalidFileName(@TempDir Path path) {
```
Comment (armiol) [inline base/src/test/java/io/spine/io/ResourceFilesTest.java]: It does not throw anything.

File: base/src/test/java/io/spine/io/ResourceFilesTest.java [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+import java.net.URL;
+import java.nio.file.Path;
+import java.util.Iterator;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+@DisplayName("Resource files should")
+@ExtendWith(TempDirectory.class)
+class ResourceFilesTest {
```
Comment (armiol) [inline base/src/test/java/io/spine/io/ResourceFilesTest.java]: Can we also have more tests for this utility? Please see how other utilities like this one are tested (tests for those usually extend `UtilityClassTest`).

File: base/src/test/java/io/spine/code/proto/DescriptorReferenceTest.java [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+import static io.spine.code.proto.given.DescriptorReferenceTestEnv.randomRef;
+import static io.spine.code.proto.given.DescriptorReferenceTestEnv.smokeTestModelCompilerRef;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Collections.emptyIterator;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@SuppressWarnings("HardcodedLineSeparator") /* Resistance to different separators is
```
Comment (armiol) [inline base/src/test/java/io/spine/code/proto/DescriptorReferenceTest.java]: Please move the suppression to the source of a warning (yes, you'll have two suppressions, but a single comment explaining both of them).

File: base/src/test/java/io/spine/code/proto/DescriptorReferenceTest.java [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+            return Iterators.singletonIterator(descRef.toUri()
+                                                      .toURL());
+        } catch (MalformedURLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @SuppressWarnings("TailRecursion")
+    // As long as the specified path does not contain files with names matching a random UUID value,
+    // recursive calls will not happen.
```
Comment (armiol) [inline base/src/test/java/io/spine/code/proto/DescriptorReferenceTest.java]: `...should not happen`.

**Comment by serhii-lekariev** [comment]
@armiol, PTAL.

**Pull request merged**